### ISSUE TITLE
pdf formatting

### ISSUE THREAD

**Issue by koerc** [koreader/koreader#8303]
KOReader version: 2021.09

Device: Kindle PW2

Issue 1
Condition:
Display pdf book; 
reflow: on;
View Mode: page;
turning from `page1` to `page2`
Issue: Start of `page2` is in most cases the start of last text paragraph of `page1`. This makes reading almost impossible.
Suggested Solution1: Start `page2` with next text line following `page1`, not paragraph 
Suggested Solution2: On `page2` grey out text already displayed on `page1`

__________________________________________________
Issue 2.1
Condition: 
Display pdf book; 
reflow: on;
View Mode: continuous;

Issue: The last text line of displayed page (page1) mostly doesnt fit on display, hence is broken.
After page turn to next page (page2) sometimes the page starts with new text line, sometimes repeats the last text line.

Suggested Solution: Cut the `page1` after last full displayed line (drop the last broken line), start next page(`page2`) with this line.

Issue 2.2
Condition: same as Issue 1.1; Status bar is displayed/not displayed
The hight of the status bar seems to be not included in the page break algorithm 

__________________________________________________
Issue 3
Condition: 
Display pdf book; 
reflow: off;
Fit: With
Issue 3.2: As one book page in this mode most likely divides in several page views, you need to change page to continue reading. Lets say book `page1` displays now as `page1.1`, followed by `page1.2`. After that comes `page2.1`. In the case we switch from `page1.1` to `page1.2`: `page1.2` ends with the correct end of the `page1`, but starts with content of `page1.1`. This makes it very hard to follow the book.
Suggested Solution1: Grey out the starting part of `page1.2` which was already displayed on `page1.1`
Suggested Solution2: Display some kind of marker on `page1.2` (dotted line?) to show where `page1.1` ended  
Suggested Solution3: Only display the new content of `page1.2` without the parts of `page1.1`
Suggested Solution4: Display content of `page1.2`, continue with content of `page2.1`

**Comment by Frenzie** [comment]
If you have separate issues, it's generally better to separate them for issue tracking reasons.

1. There's document (second icon from left) → page overlap → gray out. We'd welcome someone to help improve that, see https://github.com/koreader/koreader/issues/5520 for example. It's not the most well-tested feature since in my experience most people (or at least most developers here) use either full page or continuous, which unfortunately makes the whole overlap thing a bit unloved.
2. That's working as intended. If @Galunid is interested, it might be interesting to actually detect lines though since that should be relatively trivial using Leptonica. However, performance/battery wise it may not be desirable.
You can also adjust https://github.com/koreader/koreader/blob/feab232f35271b653778f06c80675ac527aff391/defaults.lua#L65-L66
3. See above: document (second icon from left) → page overlap → gray out.

**Comment by koerc** [comment]
Thank you very much #1 & #3 was my rtfm fault,
#2 I`ll happily try, however if line detection could be possible one day I would be double happy. 

Thanks for your great work and for this great product!!!!
